analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -106,6 +110,33 @@ private static bool TryGetTypeFromAssignmentToFloatType(SyntaxNode division, Sem
                 return type.IsAny(KnownType.NonIntegralNumbers);
             }
 
+            if (division.Ancestors().LastOrDefault(x => TupleExpressionSyntaxWrapper.IsInstance(x)) != null)
+            {
+                if (division.Ancestors().OfType<AssignmentExpressionSyntax>().FirstOrDefault() is { } assignmentSyntax
+                    && assignmentSyntax.MapAssignmentArguments() is { } assignmentMappings)
+                {
+                    var assignementLeft = assignmentMappings.Where(x => x.Right.Equals(division)).FirstOrDefault().Left;
+                    if (assignementLeft != null)
+                    {
+                        type = semanticModel.GetTypeInfo(assignementLeft).Type;
+                        return type.IsAny(KnownType.NonIntegralNumbers);
+                    }
+                }
+                else if (division.Ancestors().OfType<VariableDeclarationSyntax>().FirstOrDefault() is { } variableDeclaration)
+                {
+                    var tuple = division.Ancestors().LastOrDefault(x => TupleExpressionSyntaxWrapper.IsInstance(x));
```

Comment:
resuse `outterTuple` here.

Reply:
if `(var a, var b) = (M(1/3), 2)` `1/3` and `a` don't relate to each other. `outterTuple` must be null in this case.